tasks/main.yml

```diff
@@ -79,14 +79,35 @@
     mode: 0600
   delegate_to: "{{ ca_ca_host }}"
 
-- name: Sign CSR with CA key
+- name: Check if client certificate exists
+  ansible.builtin.stat:
+    path: "{{ ca_client_ca_dir }}/{{ inventory_hostname }}.crt"
+  register: client_certificate
+  when: ca_renew | bool
```

Comment:
I'm wondering if it would make sense to have two differrent variables. One to allow for renewing the CA and one for allowing renewing client certificates. There's a big difference in complexity and maybe you want the one but not the other.

Reply:
I do not like this, too. I will check it again and I'll try to find a better solution.

Reply:
Optimized by modularizing the tasks with tasks/renew.yml. Less tasks and vars. :-)

Reply:
Nice! That's a *really* neat trick with reusable code! (Nothing more to add, didn't close the conversation so you can read the comment.)